cmd/keys_test.go

```diff
@@ -121,3 +123,37 @@ func TestListKeys(t *testing.T) {
 		t.Errorf("Expected %v, Got %v", expected, keys)
 	}
 }
+
+type chooseKeyCases struct {
+	Reader   io.Reader
+	Expected string
+}
+
+func TestChooseKey(t *testing.T) {
+	testKeys := []api.KeyCreateRequest{
+		api.KeyCreateRequest{},
+	}
+
+	checks := []chooseKeyCases{
+		chooseKeyCases{
+			Reader:   strings.NewReader("-1"),
+			Expected: "-1 is not a valid option",
+		},
+		chooseKeyCases{
+			Reader:   strings.NewReader("2"),
+			Expected: "2 is not a valid option",
+		},
+		chooseKeyCases{
+			Reader:   strings.NewReader("a"),
+			Expected: "a is not a valid integer",
+		},
+	}
+
+	for _, check := range checks {
+		_, err := chooseKey(testKeys, check.Reader)
+
+		if err.Error() != check.Expected {
```

Comment:
tiny nit: is it possible to check on the actual error type or value, instead of the error string?